lib/phoenix/tracker/state.ex

```diff
@@ -346,19 +365,15 @@ defmodule Phoenix.Tracker.State do
   @spec remove_down_replicas(t, name) :: t
   def remove_down_replicas(%State{mode: :normal, context: ctx, values: values, pids: pids} = state, replica) do
     new_ctx = Map.delete(ctx, replica)
-    match_spec = {:_, :_, {replica, :_}}
-
-    new_clouds =
-      values
-      |> :ets.match_object(match_spec)
-      |> Enum.reduce(state.clouds, fn {{topic, pid, key}, _meta, tag}, acc ->
-        1 = :ets.select_delete(pids, [{{pid, topic, key}, [], [true]}])
-        delete_tag(acc, tag)
-      end)
-      |> Map.delete(replica)
+    # fn {{topic, pid, key}, _, {^replica, _}} -> {{pid, topic, key}, [], [true]} end
+    ms = [{{{:"$1", :"$2", :"$3"}, :_, {replica, :_}}, [], [{{{{:"$2", :"$1", :"$3"}}, [], [true]}}]}]
 
+    delete_ms = :ets.select(values, ms)
+    expected_deletes = length(delete_ms)
+    ^expected_deletes = :ets.select_delete(pids, delete_ms)
```

Comment:
Why do we do this match here? Are we trying to avoid a race?

Reply:
I just replicated the previous behaviour that would assert on each `:ets.select_delete` call - it's very likely we don't need that at all.

Reply:
I would nuke it then. :)